hw/drivers/crypto/src/crypto.c

```diff
@@ -56,8 +61,20 @@ crypto_do_ctr(struct crypto_dev *crypto, const void *key, uint16_t keylen,
             return sz + rc;
         }
 
-        for (i = 0; i < len; i++) {
-            outbuf8[i] = inbuf8[i] ^ _out[i];
+        /*
+         * For full blocks increase speed by doing 32-bit XOR; maintain the
+         * stream semantics doing byte XORs for smaller sizes (end of buffer).
+         */
+        if (len == AES_BLOCK_LEN) {
+            inbuf32 = (uint32_t *)inbuf8;
+            outbuf32 = (uint32_t *)outbuf8;
+            for (i = 0; i < len / 4; i++) {
+                outbuf32[i] = inbuf32[i] ^ _out32[i];
```

Comment:
I'm not sure what is the usage of this function but unaligned access here could crash Cortex-M0

Reply:
This should be fixed now, thanks!